msal/broker.py

```diff
@@ -151,6 +151,11 @@ def _signin_silently(
         auth_scheme=None,
         **kwargs):
     params = _build_msal_runtime_auth_params(client_id, authority)
+    if sys.platform == "linux":
+        # This is required by Linux Java Broker to set a non-empty valid redirect_uri
+        params.set_redirect_uri(
+            "https://login.microsoftonline.com/common/oauth2/nativeclient"
+        )
```

Comment:
Thanks. The change in this PR looks straightforward. In order to reduce code duplication, though, I would suggest we move the logic to a global helper function, like this:

```python
def _set_redirect_uri_for_linux(params):
    if sys.platform == "linux":
        # This is required by Linux Java Broker to set a non-empty valid redirect_uri
        params.set_redirect_uri(
            "https://login.microsoftonline.com/common/oauth2/nativeclient"
        )
```

And then the two code paths can just do:
```suggestion
    _set_redirect_uri_for_linux(params)
```

Reply:
Thanks! I just updated.